### PR TITLE
#4891 Monitor TX migration to `shared-modules`

### PR DESCRIPTION
**Pull request by desfero**
Run #p3

### CODE REVIEW

File: packages/shared-modules/src/modules/eto/sagas.spec.ts [hunk 1]
```diff
@@ -139,6 +145,18 @@ class DummyEthManager implements IEthManager {
   getWalletAddress(): Promise<EthereumAddressWithChecksum> {
     return Promise.resolve(toEthereumChecksumAddress("0x7824e49353BD72E20B61717cf82a06a4EEE209e8"));
   }
+
+  getBlockNumber(): Promise<number> {
+    throw new Error("Not implemented");
```
Comment (desfero) [inline packages/shared-modules/src/modules/eto/sagas.spec.ts:150]: Should be done as a part of #4414

File: packages/shared-modules/src/modules/tx/monitor/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    try {
+      yield* call(updatePendingTxs);
+    } catch (e) {
+      logger.error(e, "Error getting pending txs");
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+    // TODO: Add block watcher
+    yield delay(10000);
+    // yield take(actions.web3.newBlockArrived);
```
Comment (desfero) [inline packages/shared-modules/src/modules/tx/monitor/sagas.ts:202]: will do a separate issue to add block tracker to `shared-modules`

File: packages/shared-modules/src/modules/tx/monitor/sagas.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+  const isValid = schema?.isValidSync(pendingTransaction.transactionAdditionalData);
+
+  if (!isValid) {
+    throw new SchemaMismatchError(pendingTransaction.transactionType);
+  }
+}
+
+export function* txMonitorSagas(): SagaGenerator<void> {
+  yield* takeLatestUntil(authModuleAPI.actions.setUser, authModuleAPI.actions.reset, txMonitor);
```
Comment (arjunu) [inline packages/shared-modules/src/modules/tx/monitor/sagas.ts:219]: move to web app init module

**Review by arjunu — CHANGES_REQUESTED** [review]
one comment

File: packages/shared-modules/src/modules/tx/core/sagas/getTransactionOrThrow.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { call, SagaGenerator } from "@neufund/sagas";
+import { TEthereumTxHash } from "@neufund/shared-utils";
+
+import { neuGetBindings } from "../../../../utils";
```
Comment (ozanmanav-ng) [inline packages/shared-modules/src/modules/tx/core/sagas/getTransactionOrThrow.ts:4]: Shouldn't we use alias for this ?

Comment (desfero) [reply packages/shared-modules/src/modules/tx/core/sagas/getTransactionOrThrow.ts:4]: good point, unfortunately we don't have yet setup for aliases in `shared-modules`

File: packages/shared-modules/src/modules/tx/monitor/sagas.ts (hunk 3 above)
Comment (desfero) [reply packages/shared-modules/src/modules/tx/monitor/sagas.ts]: done

**Review by arjunu — APPROVED** [review]

**Pull request merged**